### ISSUE TITLE
error: no match for "centos/ruby-22-centos7"

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#7490]
Import of an image is suddenly failing?

```
0220 00:05:58.486923   46250 dockerimagelookup.go:72] checking remote registry for "centos/ruby-22-centos7"
I0220 00:05:58.487538   46250 request.go:579] Request Body: {"kind":"ImageStreamImport","apiVersion":"v1","metadata":{"name":"newapp","creationTimestamp":null},"spec":{"import":false,"images":[{"from":{"kind":"DockerImage","name":"centos/ruby-22-centos7"},"importPolicy":{}}]},"status":{}}
I0220 00:05:58.487575   46250 round_trippers.go:264] POST https://10.1.2.2:8443/oapi/v1/namespaces/foo/imagestreamimports
I0220 00:05:58.487587   46250 round_trippers.go:271] Request Headers:
I0220 00:05:58.487590   46250 round_trippers.go:274]     Content-Type: application/json
I0220 00:05:58.487593   46250 round_trippers.go:274]     Accept: application/json, */*
I0220 00:05:58.487596   46250 round_trippers.go:274]     User-Agent: oc/v1.1.3 (darwin/amd64) openshift/1353b8c
I0220 00:05:59.836300   46250 round_trippers.go:289] Response Status: 201 Created in 1348 milliseconds
I0220 00:05:59.836320   46250 round_trippers.go:292] Response Headers:
I0220 00:05:59.836324   46250 round_trippers.go:295]     Cache-Control: no-store
I0220 00:05:59.836327   46250 round_trippers.go:295]     Content-Type: application/json
I0220 00:05:59.836330   46250 round_trippers.go:295]     Date: Fri, 19 Feb 2016 01:03:12 GMT
I0220 00:05:59.836803   46250 request.go:875] Response Body: {"kind":"ImageStreamImport","apiVersion":"v1","metadata":{"name":"newapp","namespace":"foo","selfLink":"/oapi/v1/namespaces/foo/imagestreamimports/newapp","uid":"8bd9476b-d6a4-11e5-adac-080027c5bfa9","creationTimestamp":"2016-02-19T01:03:11Z"},"spec":{"import":false,"images":[{"from":{"kind":"DockerImage","name":"centos/ruby-22-centos7"},"importPolicy":{}}]},"status":{"images":[{"status":{"metadata":{},"status":"Success"},"image":{"metadata":{"name":"sha256:7b215202d0203ab3f7c9e76033361ce98541d0b21df585bce2fa97d2cb8776b0","creationTimestamp":null},"dockerImageReference":"centos/ruby-22-centos7@sha256:7b215202d0203ab3f7c9e76033361ce98541d0b21df585bce2fa97d2cb8776b0","dockerImageMetadata":{"kind":"DockerImage","apiVersion":"1.0","Id":"cf5380713f1e6bc5cec66dc0d50b10e8f5369e1ff7490f4840d7fdcc19b6d115","Parent":"422ce95c2ff7abe1178aa7be47ec20357bc2cf0a94ea796ee2fb8dead375b696","Created":"2016-02-19T22:02:05Z","Container":"68ca5e87f4f9d335d37f1484f586c602db2855a6365d6e76a6038776780ec748","ContainerConfig":{"Hostname":"82f7a8302f48","User":"1001","ExposedPorts":{"8080/tcp":{}},"Env":["PATH=/opt/app-root/src/bin:/opt/app-root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","STI_SCRIPTS_URL=image:///usr/libexec/s2i","STI_SCRIPTS_PATH=/usr/libexec/s2i","HOME=/opt/app-root/src","BASH_ENV=/opt/app-root/etc/scl_enable","ENV=/opt/app-root/etc/scl_enable","PROMPT_COMMAND=. /opt/app-root/etc/scl_enable","RUBY_VERSION=2.2"],"Cmd":["/bin/sh","-c","#(nop) LABEL io.openshift.builder-version=8847438ba06307f86ac877465eadc835201241df"],"Image":"422ce95c2ff7abe1178aa7be47ec20357bc2cf0a94ea796ee2fb8dead375b696","WorkingDir":"/opt/app-root/src","Entrypoint":["container-entrypoint"],"Labels":{"build-date":"2016-02-17","io.k8s.description":"Platform for building and running Ruby 2.2 applications","io.k8s.display-name":"Ruby 2.2","io.openshift.builder-base-version":"88b0fe5","io.openshift.builder-version":"8847438ba06307f86ac877465eadc835201241df","io.openshift.expose-services":"8080:http","io.openshift.s2i.scripts-url":"image:///usr/libexec/s2i","io.openshift.tags":"builder,ruby,ruby22","io.s2i.scripts-url":"image:///usr/libexec/s2i","license":"GPLv2","name":"CentOS Base Image","vendor":"CentOS"}},"DockerVersion":"1.8.2-el7","Author":"SoftwareCollections.org \u003csclorg@redhat.com\u003e","Config":{"Hostname":"82f7a8302f48","User":"1001","ExposedPorts":{"8080/tcp":{}},"Env":["PATH=/opt/app-root/src/bin:/opt/app-root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","STI_SCRIPTS_URL=image:///usr/libexec/s2i","STI_SCRIPTS_PATH=/usr/libexec/s2i","HOME=/opt/app-root/src","BASH_ENV=/opt/app-root/etc/scl_enable","ENV=/opt/app-root/etc/scl_enable","PROMPT_COMMAND=. /opt/app-root/etc/scl_enable","RUBY_VERSION=2.2"],"Cmd":["/bin/sh","-c","$STI_SCRIPTS_PATH/usage"],"Image":"422ce95c2ff7abe1178aa7be47ec20357bc2cf0a94ea796ee2fb8dead375b696","WorkingDir":"/opt/app-root/src","Entrypoint":["container-entrypoint"],"Labels":{"build-date":"2016-02-17","io.k8s.description":"Platform for building and running Ruby 2.2 applications","io.k8s.display-name":"Ruby 2.2","io.openshift.builder-base-version":"88b0fe5","io.openshift.builder-version":"8847438ba06307f86ac877465eadc835201241df","io.openshift.expose-services":"8080:http","io.openshift.s2i.scripts-url":"image:///usr/libexec/s2i","io.openshift.tags":"builder,ruby,ruby22","io.s2i.scripts-url":"image:///usr/libexec/s2i","license":"GPLv2","name":"CentOS Base Image","vendor":"CentOS"}},"Architecture":"amd64","Size":441434825},"dockerImageMetadataVersion":"1.0","dockerImageLayers":[{"name":"sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4","size":0},{"name":"sha256:ed99f9270efc5694a6f5a6fd276421a8b01ca1cd5c00c2d3e50a8068a8e16a1c","size":196610919},{"name":"sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4","size":0},{"name":"sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:9a1c49180f4542d876f3f98b3d02cabdd86c97bff31d7a6bde779b65d5fdbf22","size":81},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:c847b3477096385877c217a4e7b7953ce93383a869f7e8f7f99470cbb686283d","size":187628596},{"name":"sha256:9b02e4d83917cc0654a6aa34e3e498b2c6cf29918bc940d3dd6cd48a243cafc2","size":913},{"name":"sha256:3fd92105dbb33cd9f34bb004f079701668cabf9dcc48c73e07385c501be8a2c5","size":53},{"name":"sha256:cfce03235c331b87d5f9f49e52b238a4fffa7e23a5b9cabba75ff230c24a3c90","size":193},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:f4e52d7c7380d3dd42296b0a307210eed0ff5af98f61d33d07098ff89113ae05","size":57189623},{"name":"sha256:b20fcaa085a77ad3d82f5cb02c0174ad0d7d841f08c2bb7d9026277e9e6c7187","size":3269},{"name":"sha256:099cb6bdbe8e42ee4c96019ad3082ff48a93692a181005e3e29dd82b288f8c26","size":589},{"name":"sha256:97c79772bdf028020051f7d04384bb19665ae2fb2109774be5a498f3ab6e6c36","size":589},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0},{"name":"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef","size":0}]},"tag":"latest"}]}}
I0220 00:05:59.838835   46250 dockerimagelookup.go:237] Adding Docker image "centos/ruby-22-centos7" (tag "latest"), cf53807, from Docker Hub, 420.985mb, author SoftwareCollections.org <sclorg@redhat.com> as component match for "centos/ruby-22-centos7" with score 0
I0220 00:05:59.839573   46250 dockerimagelookup.go:78] Found remote match centos/ruby-22-centos7
F0220 00:05:59.839661   46250 helpers.go:107] error: no match for "centos/ruby-22-centos7"

The 'oc new-app' command will match arguments to the following types:

  1. Images tagged into image streams in the current project or the 'openshift' project
     - if you don't specify a tag, we'll add ':latest'
  2. Images in the Docker Hub, on remote registries, or on the local Docker engine
  3. Templates in the current project or the 'openshift' project
  4. Git repository URLs or local paths that point to Git repositories

--allow-missing-images can be used to point to an image that does not exist yet.

See 'oc new-app -h' for examples.
```

Looks like it finds the image fine.

**Comment by pweil-** [comment]
cc @miminar

**Comment by soltysh** [comment]
@smarterclayton what's the command you've used to invoke it? I've been playing with oc new-app and that specific image this morning, tried once again just now and it all works ok on my end.

**Comment by smarterclayton** [comment]
I fixed this over the weekend with #7499

**Comment by soltysh** [comment]
Thank you!
